packages/e2e/test/index.ts

```diff
@@ -34,4 +36,44 @@ describe("e2e tests", function () {
       assert.equal(hhCleanCode2, 0);
     });
   });
+
+  describe("sample project", function () {
+    // These tests generate the sample project and then exercise the commands
+    // that are suggested to the user after project generation.  It would be
+    // better if that list of commands were exernalized somewhere, in a place
+    // from which we could consume them here, so that the list of commands
+    // executed here cannot fall out of sync with what's actually suggested to
+    // the user, but this approach was more expedient.
+
+    // Also, it would be better if we had separate, independent tests for each
+    // of the individual suggested commands, but because the project creation
+    // takes so long (really it's the dependency installation that's so time
+    // consuming) even a single test already has a crazy long run time, so for
+    // expediency this is one big complex test.
+
+    useFixture("sample-project");
+
+    it("should permit successful execution all of the suggested commands", async function () {
+      if (os.type() === "Windows_NT") {
+        // cf. https://github.com/nomiclabs/hardhat/issues/1698
+        this.skip();
+      }
+      try {
+        shell.exec(`${hardhatBinary}`, {
+          env: {
+            ...process.env,
+            HARDHAT_CREATE_SAMPLE_PROJECT_WITH_DEFAULTS: "true",
+          },
+        });
+        shell.exec(`${hardhatBinary} accounts`);
+        shell.exec(`${hardhatBinary} compile`);
+        shell.exec(`${hardhatBinary} test`);
+        shell.exec("node scripts/sample-script.js");
```

Comment:
I think these commands are actually not working. The reason the tests pass anyway is that `shelljs`, by default, doesn't throw when a command run with `shell.exec` exits with a non-zero code. I don't blame you for not realizing this, it's bitten me more times than I can remember :sweat: 

The quick fix here is to use `shell.set("-e");` at the beginning. I don't really like this approach because it feels like a global thing (my guess is that changes the whole module behavior and I don't understand the implications of that). The alternative is to do what I do in the previous tests and assert the value of the exit code.

Reply:
They actually were working, but I changed one to fail, and you're right, failures were not being handled correctly.  Addressed in https://github.com/nomiclabs/hardhat/pull/1686/commits/df3ed5454a55dd21bdfc1d44564caf790da36c68